x/incentives/keeper/gauge.go

```diff
@@ -110,13 +123,60 @@ func (k Keeper) CreateGauge(ctx sdk.Context, isPerpetual bool, owner sdk.AccAddr
 		}
 	}
 
-	// check if denom this gauge pays out to exists on-chain
-	if !k.bk.HasSupply(ctx, distrTo.Denom) && !strings.Contains(distrTo.Denom, "osmovaloper") {
-		return 0, fmt.Errorf("denom does not exist: %s", distrTo.Denom)
+	nextGaugeId := k.GetLastGaugeID(ctx) + 1
+
+	// For no lock gauges, a pool id must be set.
+	// A pool with such id must exist and be a concentrated pool.
+	if distrTo.LockQueryType == lockuptypes.NoLock {
+		if poolId == 0 {
+			return 0, fmt.Errorf("'no lock' type gauges must have a pool id")
+		}
+
+		// If not internal gauge denom, then must be set to ""
+		// and get overwritten with the external prefix + pool id
+		// for internal query purposes.
+		distrToDenom := distrTo.Denom
+		if distrToDenom != types.NoLockInternalGaugeDenom(poolId) {
+			// If denom is set, then fails.
+			if distrToDenom != "" {
+				return 0, fmt.Errorf("'no lock' type external gauges must have an empty denom set, was %s", distrToDenom)
+			}
+			distrTo.Denom = types.NoLockExternalGaugeDenom(poolId)
+		}
+
+		pool, err := k.pmk.GetPool(ctx, poolId)
+		if err != nil {
+			return 0, err
+		}
+
+		if pool.GetType() != poolmanagertypes.Concentrated {
+			return 0, fmt.Errorf("'no lock' type gauges must be created for concentrated pools only")
+		}
+
+		// Note that this is a general linking between the gauge and the pool
+		// for "NoLock" gauges. It occurs for both external and internal gauges.
+		// That being said, internal gauges have an additional linking
+		// by duration where duration is the incentives epoch duration.
+		// The internal incentive linking is set in x/pool-incentives CreateConcentratedLiquidityPoolGauge.
+		k.pik.SetPoolGaugeIdNoLock(ctx, poolId, nextGaugeId)
+	} else {
```

Comment:
nit: i think we should just explicitly call the duration lock type here, and then the else branch errors.

Reply:
This validation is specific to "ByDuration" gauges. The logic below the if/else is shared between "NoLock" and "ByDuration". As a result, I don't think this suggestion is possible